src/Tribe/API.php

```diff
@@ -602,5 +642,222 @@ public static function get_and_flatten_event_meta( $event_id ) {
+				|| Tribe__Events__Main::POSTTYPE !== $args['post_type']
+				|| ! is_array( $args )
+			) {
+				return $args;
+			}
+
+			// Sanitize if valid, or fail with WP_Error if invalid.
+			// Could enhance this with more comprehensive checks in the future.
+
+			// Process meridian fields before hour fields to determine if hours are to be in the 12-hour or 24-hour format.
+			if ( ! empty( $args['EventStartMeridian'] ) ) {
+				$args['EventStartMeridian'] = self::sanitize_meridian_meta_value( $args['EventStartMeridian'] );
+			}
+
+			if ( ! empty( $args['EventEndMeridian'] ) ) {
+				$args['EventEndMeridian'] = self::sanitize_meridian_meta_value( $args['EventEndMeridian'] );
+			}
+
+			// If meridian is set but we can pretty easily guess the hour is a valid 24-hour format, discard meridian in attempt to be smarter/flexible, thus setting "14pm" (invalid) to "14" (equivalent to 2pm).
+			// We take this approach instead of just minus `12` from the integer value of the hour because meridian may be set by defaults.
+			// Watch out for sending "12pm" or "12am" as the meridian (12-hour format) will take precedence if it exists.
+			if (
+				! empty( $args['EventStartMeridian'] )
+				&& 12 < absint( $args['EventStartHour'] )
+				&& 24 > absint( $args['EventStartHour'] )
+			) {
+				$args['EventStartMeridian'] = '';
+			}
+
+			if (
+				! empty( $args['EventEndMeridian'] )
+				&& 12 < absint( $args['EventEndHour'] )
+				&& 24 > absint( $args['EventEndHour'] )
+			) {
+				$args['EventEndMeridian'] = '';
+			}
+
+			// Now process all but the meridians
+			foreach ( $args as $key => &$value ) {
+				if ( 'EventStartHour' === $key ) {
+					if ( empty( $args['EventStartMeridian'] ) ) {
+						$value = self::sanitize_hour_meta_value( $value );
+					} else {
+						$value = self::sanitize_hour_meta_value( $value, true );
+					}
+				} elseif ( 'EventEndHour' === $key ) {
+					if ( empty( $args['EventEndMeridian'] ) ) {
+						$value = self::sanitize_hour_meta_value( $value );
+					} else {
+						$value = self::sanitize_hour_meta_value( $value, true );
+					}
+				} elseif (
+					'EventStartMinute' === $key
+					|| 'EventEndMinute' === $key
+				) {
+					$value = self::sanitize_minute_meta_value( $value );
+				}
+
+				if ( is_wp_error( $value ) ) {
+					return $value;
+				}
+			}
+
+			return $args;
+		}
+
+		/**
+		 * Sanitize a string to be used as an event meridian post meta value: am|pm.
+		 *
+		 * Use this prior to sending value to the database.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $value            The post meta value to be checked, such as 'am'.
+		 * @param bool   $empty_if_invalid If true, set an invalid value to an empty string, else generate WP_Error.
+		 *
+		 * @return string|WP_Error
+		 */
+		public static function sanitize_meridian_meta_value( $value, $empty_if_invalid = true ) {
+			// lower-case to match the `a` PHP date format used elsewhere
+			$new_value = strtolower( trim( $value ) );
+
+			if (
```

Comment:
To avoid nesting I think you can refactor this one into something like this:

```php
$is_valid = 'am' === $new_value || 'pm' === $new_value;
$invalid_value = '';

if ( ! $empty_if_invalid ) {
	$invalid_value = new WP_Error(...);
}

return $is_valid ? $new_value : $invalid_value;
```